lib/puppet/provider/user/aix.rb

```diff
@@ -255,7 +257,11 @@ def password=(value)
     user = @resource[:name]
 
     # Puppet execute does not support strings as input, only files.
-    tmpfile = Tempfile.new('puppet_#{user}_pw')
+    # The password is expected to be in an encrypted format given -e is specified:
+    # https://www.ibm.com/support/knowledgecenter/ssw_aix_71/com.ibm.aix.cmds1/chpasswd.htm
+    # /etc/security/passwd is specified as an ASCII file per the AIX documentation
+    # TODO: Is there a reason we don't simply pipe the output to chpasswd like `echo foo | chpasswd`
+    tmpfile = Tempfile.new('puppet_#{user}_pw', :encoding => Encoding::ASCII)
```

Comment:
Yeah, I don't even want to know how far back the `git blame` on that goes :). Would be nice to get it fixed up while we're in here, though.